OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -2620,6 +2624,44 @@ void onAnswered(OneSignal.PromptActionResult result) {
 
       runPromptLocation.run();
    }
+   
+   public static void getOneSignalNotificationsCount(final GetNotificationsCountHandler getNotificationsCountHandler) {
+      Runnable runGetOneSignalNotificationsCount = new Runnable() {
+         @Override
+         public void run() {
+            NotificationManager notificationManager = OneSignalNotificationManager.getNotificationManager(appContext);
+
+            OneSignalDbHelper dbHelper = OneSignalDbHelper.getInstance(appContext);
+            String[] retColumn = {OneSignalDbContract.NotificationTable.COLUMN_NAME_ANDROID_NOTIFICATION_ID};
+
+            Cursor cursor = dbHelper.query(
+                    OneSignalDbContract.NotificationTable.TABLE_NAME,
+                    retColumn,
+                    OneSignalDbContract.NotificationTable.COLUMN_NAME_DISMISSED + " = 0 AND " +
+                            OneSignalDbContract.NotificationTable.COLUMN_NAME_OPENED + " = 0",
+                    null,
+                    null,                                                    // group by
+                    null,                                                    // filter by row groups
+                    null                                                     // sort order
+            );
+            
+            //BadgeCountUpdater.updateCount(0, appContext);			
+			getNotificationsCountHandler.notificationsAvailable(cursor.getCount());
+
+            cursor.close();
+         }
+      };
+
+      if (appContext == null || shouldRunTaskThroughQueue()) {
+         Log(LOG_LEVEL.ERROR, "OneSignal.init has not been called. " +
+                 "Could not get notifications count at this time - moving this operation to" +
+                 "a waiting task queue.");
+         addTaskToQueue(new PendingTaskRunnable(runGetOneSignalNotificationsCount));
```

Comment:
Also since we now wait on RemoteParam you might want to check how is done on major release now.

Reply:
Failed to build an operational version of the lib based on the recent commit from master (either from 4.1.0 tag) locally. Also, please, note - the current version of a Unity wrapper is not synced with the changes in the com.onesignal package.